boa_engine/src/builtins/array/mod.rs

```diff
@@ -236,6 +236,27 @@ impl BuiltInConstructor for Array {
 }
 
 impl Array {
+    /// Optimized helper function, that sets the length of the array.
+    fn set_length(o: &JsObject, len: u64, context: &mut Context<'_>) -> JsResult<()> {
+        if o.is_array() && len < (2u64.pow(32) - 1) {
+            let mut borrowed_object = o.borrow_mut();
+            if borrowed_object.properties().shape.to_addr_usize()
+                == context
+                    .intrinsics()
+                    .templates()
+                    .array()
+                    .shape()
+                    .to_addr_usize()
+            {
+                // NOTE: The "length" property is the first element.
+                borrowed_object.properties_mut().storage[0] = JsValue::new(len);
+                return Ok(());
+            }
+        }
+
+        o.set(utf16!("length"), len, true, context).map(|_| ())
```

Comment:
I think it's better to use `?` here and return `Ok(())`. Just makes it a bit more readable